internal/pkg/stream/ecs.go

```diff
@@ -157,11 +168,27 @@ func (s *ECSDeploymentStreamer) Fetch() (next time.Time, done bool, err error) {
 		}
 		s.pastEventIDs[id] = true
 	}
+
+	var alarms []cloudwatch.AlarmStatus
+	if out.DeploymentConfiguration != nil && out.DeploymentConfiguration.Alarms != nil && aws.BoolValue(out.DeploymentConfiguration.Alarms.Enable) {
+		alarmNames := aws.StringValueSlice(out.DeploymentConfiguration.Alarms.AlarmNames)
+		alarms, err = s.cw.AlarmStatuses(cloudwatch.WithNames(alarmNames))
+		if err != nil {
+			if request.IsErrorThrottle(err) {
+				s.cwRetries += 1
+				return nextFetchDate(s.clock, s.rand, s.cwRetries), false, nil
+			}
+			return next, false, fmt.Errorf("retrieve alarm statuses: %w", err)
+		}
+		s.cwRetries = 0
+	}
+	
 	s.eventsToFlush = append(s.eventsToFlush, ECSService{
 		Deployments:         deployments,
 		LatestFailureEvents: failureMsgs,
+		Alarms:              alarms,
 	})
-	return nextFetchDate(s.clock, s.rand, s.retries), done, nil
+	return nextFetchDate(s.clock, s.rand, s.ecsRetries), done, nil
```

Comment:
nit: This way folks might not be confused why this is only `ecsRetries` and not `cwRetries`
```suggestion
	return nextFetchDate(s.clock, s.rand, 0), done, nil
```